# Serie 11
## Aufgabe 11.1
In dieser Aufgabe verwenden wir den Datensatz Auto, der in der Bibliothek `ISLR` enthalten ist.

In [3]:
# install.packages("ISLR") # only once
library(ISLR)

a) Untersuchen Sie den Datensatz mit `head(Auto)` und `?Auto`.

In [4]:
head(Auto)
?Auto

mpg cylinders displacement horsepower weight acceleration year origin
1 18  8         307          130        3504   12.0         70   1     
2 15  8         350          165        3693   11.5         70   1     
3 18  8         318          150        3436   11.0         70   1     
4 16  8         304          150        3433   12.0         70   1     
5 17  8         302          140        3449   10.5         70   1     
6 15  8         429          198        4341   10.0         70   1     
  name                     
1 chevrolet chevelle malibu
2 buick skylark 320        
3 plymouth satellite       
4 amc rebel sst            
5 ford torino              
6 ford galaxie 500

Auto                   package:ISLR                    R Documentation

_A_u_t_o _D_a_t_a _S_e_t

_D_e_s_c_r_i_p_t_i_o_n:

     Gas mileage, horsepower, and other information for 392 vehicles.

_U_s_a_g_e:

     Auto
     
_F_o_r_m_a_t:

     A data frame with 392 observations on the following 9 variables.

     ‘mpg’ miles per gallon

     ‘cylinders’ Number of cylinders between 4 and 8

     ‘displacement’ Engine displacement (cu. inches)

     ‘horsepower’ Engine horsepower

     ‘weight’ Vehicle weight (lbs.)

     ‘acceleration’ Time to accelerate from 0 to 60 mph (sec.)

     ‘year’ Model year (modulo 100)

     ‘origin’ Origin of car (1. American, 2. European, 3. Japanese)

     ‘name’ Vehicle name

     The orginal data contained 408 observations but 16 observations
     with missing values were removed.

_S_o_u_r_c_e:

     This dataset was taken from the StatLib library which is
     maintained at Carnegie Mellon University. The dataset was used in
     the 1983 American Statistical Association Exposition.

_R_e_f_e_r_e_n_c_e_s:

     James, G., Witten, D., Hastie, T., and Tibshirani, R. (2013) _An
     Introduction to Statistical Learning with applications in R_,
     <URL: www.StatLearning.com>, Springer-Verlag, New York

_E_x_a_m_p_l_e_s:

     pairs(Auto)
     attach(Auto)
     hist(mpg)

b) Stellen Sie das Modell für eine einfache lineare Regression mit mpg als Zielvariable und horsepower als Prädiktor auf.


$mpg \approx \beta_0 + \beta_1 * horsepower$

c) Verwenden Sie den `lm()`-Befehl um diese Regression durchzuführen.

In [24]:
lmMH <- lm(Auto$mpg ~ Auto$horsepower)
# alternative: lm(mpg ~ horsepower, data = Auto)
lmMH


Call:
lm(formula = Auto$mpg ~ Auto$horsepower)

Coefficients:
    (Intercept)  Auto$horsepower  
        39.9359          -0.1578  


In [25]:
summary(lmMH)


Call:
lm(formula = Auto$mpg ~ Auto$horsepower)

Residuals:
     Min       1Q   Median       3Q      Max 
-13.5710  -3.2592  -0.3435   2.7630  16.9240 

Coefficients:
                 Estimate Std. Error t value Pr(>|t|)    
(Intercept)     39.935861   0.717499   55.66   <2e-16 ***
Auto$horsepower -0.157845   0.006446  -24.49   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 4.906 on 390 degrees of freedom
Multiple R-squared:  0.6059,	Adjusted R-squared:  0.6049 
F-statistic: 599.7 on 1 and 390 DF,  p-value: < 2.2e-16


i) Gibt es einen Zusammenhang zwischen der Zielgrösse und dem Prädiktor?

Ja der $p$-Wert ist fast 0 - die Nullhypthese ($\beta_1=0$) wird verworfen; je mehr Pferdestärken ein Auto hat, umso weniger weit kann man fahren pro Gallone.

ii) Wie interpretieren Sie die Koeffizienten für `intercept` und `horsepower`? Ist der Zusammenhang positiv oder negativ?

Der Intercept von 39.94 für 0PS ist natürlich hier unsinnig. Die Steigung ist negativ, weil die Reichweite durch die Zunahme der Pferdestärken abnimmt.

iii) Bestimmen Sie die Vertrauensintervalle (mit `confint()`) und interpretieren Sie diese?

In [26]:
confint(lmMH, level=0.95)

2.5 %     97.5 %    
(Intercept)     38.525212 41.3465103
Auto$horsepower -0.170517 -0.1451725

Für das 95%-Vertrauensintervall $\beta_0$ gilt [38.525,41.347] und $\beta_1$ gilt [-0.171,-0.145]. Die Intervalle sind recht *schmal*, weshalb die Aussagekraft relativ gross ist.

iv) Interpretieren Sie den $R^2$-Wert.

In [28]:
#names(summary(lmMH)) # names gibt output parameter zurück, welche via $ abgefragt werden können
summary(lmMH)$r.squared

[1] 0.6059483

Der $R^2$-Wert ist zwar näher bei 1 als bei 0, beschreibt das Modell aber nicht ganz zuverlässig. Er gibt an, dass die Variabilität zu 60% durch das Modell erklärt wird.